### PR TITLE
Enable setting forward error correction for an interface

### PR DESCRIPTION
**Pull request by sc68cal**
This adds the ability to enable or disable FEC on interfaces managed via the `os10_interface` role

### CODE REVIEW

File: roles/os10_interface/templates/os10_interface.j2 [hunk 1]
```diff
@@ -108,6 +108,12 @@ interface {{ interface_key }}
     {% endif %}
   {% endif %}
 
+  {% if intf_vars.fec is defined %}
+    {% if intf_vars.fec %}
+ fec {{ intf_vars.fec }}
```
Comment (gunasekaran-t) [inline roles/os10_interface/templates/os10_interface.j2:113]: Please add 'no' form of the CLI as well

**Comment by gunasekaran-t** [comment]
Please add a changelog fragment file

**Comment by gunasekaran-t** [comment]
Please update the document file as well.

**Comment by Andersson007** [comment]
@sc68cal hello, how about adding the fragment (see https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment) and updating the documentation?

**Comment by sc68cal** [comment]
> 
> 
> @sc68cal hello, how about adding the fragment (see https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment) and updating the documentation?

Yeah I'll add a changelog fragment and update the docs next week.

**Comment by Andersson007** [comment]
Thanks! Could you also please mark this as not a draft if it isn't?

File: changelogs/fragments/forward-error-correction-feature.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+---
+minor_changes:
+- add support for setting FEC on interfaces
```
Comment (Andersson007) [inline changelogs/fragments/forward-error-correction-feature.yaml:3]:
```suggestion
- Add support for setting FEC on interfaces (https://github.com/ansible-collections/dellemc.os10/pull/78)
```

Comment (sc68cal) [reply changelogs/fragments/forward-error-correction-feature.yaml:3]: My concern is that adding the URL for the pull request, creates a circular dependency. You must create a PR in github before you can create the changelog fragment, which makes things inconvenient.

Comment (Andersson007) [reply changelogs/fragments/forward-error-correction-feature.yaml:3]: yeah, it's inconvenient. You could create an issue (which is preferable) that the PR fixes and put its URL in the fragment.

Comment (sc68cal) [reply changelogs/fragments/forward-error-correction-feature.yaml]: Fine. I'll open an issue.

**Review by Andersson007 — APPROVED** [review]

**Comment by Andersson007** [comment]
@gowsalya-mariappan hello, would you like to suggest changes or merge this PR or I could?

File: changelogs/fragments/forward-error-correction-feature.yaml (hunk 2 above)
Comment (Andersson007) [reply changelogs/fragments/forward-error-correction-feature.yaml]: If no response from the maintainers, I'll merge this tomorrow morning

**Comment by Andersson007** [comment]
@sc68cal thanks for the improvement! @gunasekaran-t thanks for reviewing!

**Pull request merged**